javascript/ql/src/semmle/javascript/PackageExports.qll

```diff
@@ -70,6 +70,61 @@ private DataFlow::Node getAValueExportedByPackage() {
     result = cla.getAStaticMethod() or
     result = cla.getConstructor()
   )
+  or
+  // *****
+  // Various standard library methods for transforming exported objects.
+  // *****
+  //
+  // Object.defineProperties
+  exists(DataFlow::MethodCallNode call |
+    call = DataFlow::globalVarRef("Object").getAMethodCall("defineProperties") and
+    [call, call.getArgument(0)] = getAValueExportedByPackage() and
+    result = call.getArgument(any(int i | i > 0))
+  )
+  or
+  // Object.defineProperty
+  exists(DataFlow::MethodCallNode call |
+    call = DataFlow::globalVarRef("Object").getAMethodCall("defineProperty") and
+    [call, call.getArgument(0)] = getAValueExportedByPackage()
+  |
+    result = call.getArgument(2).getALocalSource().getAPropertyReference("value")
+    or
+    result =
+      call.getArgument(2)
+          .getALocalSource()
+          .getAPropertyReference("get")
+          .(DataFlow::FunctionNode)
+          .getAReturn()
+  )
+  or
+  // Object.assign
+  exists(DataFlow::MethodCallNode assign |
+    assign = DataFlow::globalVarRef("Object").getAMethodCall("assign")
```

Comment:
Huh. My comment about `ExtendCall` got lost.

Another case of potential reuse: https://github.com/github/codeql/blob/746e9948b07f730c8c0a89e6ebcac5fa300b9d30/javascript/ql/src/semmle/javascript/Extend.qll#L11